tests/unit/test_charm.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Synapse charm unit tests."""
+
+from secrets import token_hex
+
+import ops
+import pytest
+from ops.testing import Harness
+
+import synapse
+from charm_state import SYNAPSE_CONTAINER_NAME
+
+
+def test_synapse_pebble_layer(harness: Harness) -> None:
+    """
+    arrange: none
+    act: start the Synapse charm, set Synapse container to be ready and set server_name.
+    assert: Synapse charm should submit the correct Synapse pebble layer to pebble.
+    """
+    harness.begin_with_initial_hooks()
+    harness.set_can_connect(harness.model.unit.containers[SYNAPSE_CONTAINER_NAME], True)
+    harness.disable_hooks()
+    server_name = token_hex(16)
```

Comment:
Perhaps set it to something that identifies the test?

Reply:
Hum...I'm confused, David. Isn't that the same case that you suggested [here](https://github.com/canonical/indico-operator/pull/248#discussion_r1139815154) to use token_hex instead of a value?